File: app/services/intent.py
"""
Intent classification for inbound WhatsApp messages.

MVP: fast, free, zero-latency keyword matching — good enough for the
core intents these businesses actually need.

UPGRADE PATH (post-revenue, when volume/nuance justifies the cost):
swap `classify()` internals to call the Anthropic API with a small
prompt like:
    "Classify this message from a customer of a {niche} business into
     one of: quote, booking, pricing, callback, complaint, other.
     Message: {text}. Respond with just the label."
Keep the function signature identical so nothing else in the app changes.
"""

GREETINGS = {"hi", "hello", "hey", "start", "menu", "sawubona", "dumela", "lumela", "thobela"}
QUOTE_WORDS = {"quote", "price", "how much", "cost", "estimate"}
BOOK_WORDS = {"book", "booking", "appointment", "schedule", "when can you", "reserve"}
CALLBACK_WORDS = {"call me", "call back", "phone me", "ring me", "call"}
PRICING_WORDS = {"prices", "pricing", "rates", "packages", "how much does it cost"}
NEGATION = {"don't", "do not", "no need", "not needed", "not interested", "just checking"}


def classify(text: str) -> str:
    t = (text or "").strip().lower()

    if any(t == g or t.startswith(g) for g in GREETINGS):
        return "greeting"
    if any(n in t for n in NEGATION):
        return "other"
    if any(w in t for w in BOOK_WORDS):
        return "booking"
    if any(w in t for w in CALLBACK_WORDS):
        return "callback"
    if any(w in t for w in PRICING_WORDS):
        return "pricing"
    if any(w in t for w in QUOTE_WORDS):
        return "quote"
    return "other"

File: src/modules/messages/intent.ts
/**
 * Keyword-based intent classifier. Same deliberate MVP choice as the
 * previous FastAPI version: zero cost, zero latency, "good enough to
 * route" rather than perfect. Upgrade path (documented, not built):
 * swap the body of `classify()` for a call through the AI provider
 * abstraction (architecture standard §3) — keep the signature identical
 * so callers never need to change.
 */

const GREETINGS = ["hi", "hello", "hey", "start", "menu", "sawubona", "dumela"];
const BOOK_WORDS = ["book", "booking", "appointment", "schedule"];
const CALLBACK_WORDS = ["call me", "call back", "phone me", "ring me"];
const PRICING_WORDS = ["prices", "pricing", "rates", "packages"];
const QUOTE_WORDS = ["quote", "price", "how much", "cost", "estimate"];

export type Intent = "greeting" | "booking" | "callback" | "pricing" | "quote" | "other";

export function classifyIntent(text: string): Intent {
  const t = (text ?? "").trim().toLowerCase();

  if (GREETINGS.some((g) => t === g || t.startsWith(g))) return "greeting";
  if (BOOK_WORDS.some((w) => t.includes(w))) return "booking";
  if (CALLBACK_WORDS.some((w) => t.includes(w))) return "callback";
  if (PRICING_WORDS.some((w) => t.includes(w))) return "pricing";
  if (QUOTE_WORDS.some((w) => t.includes(w))) return "quote";
  return "other";
}

export const REPLIES: Record<Intent, string> = {
  greeting: "Hi! Reply with one of these:\n1) Quote\n2) Book\n3) Prices\n4) Call back",
  quote: "Great — send us: service needed, suburb, and preferred time. We'll quote you today.",
  booking: "Let's get you booked. What day and time works best, and what's the job?",
  pricing: "Our starting rates depend on the job — tell us what you need and we'll send exact pricing.",
  callback: "Got it — we'll call you back shortly. What's the best time to reach you?",
  other: "Thanks for your message! Reply MENU to see options.",
};
